wp-content/plugins/wporg-learn/inc/sensei.php

```diff
@@ -163,3 +163,81 @@ function wporg_fix_learning_mode_header_space() {
 	wp_add_inline_style( 'learning-mode-header-fix', $custom_styles );
 }
 add_action( 'sensei_course_learning_mode_load_theme', __NAMESPACE__ . '\wporg_fix_learning_mode_header_space' );
+
+/**
+ * Format a date query var into a DateTime object.
+ */
+function wporg_learn_get_date( $query_var ) {
+	$date = sanitize_text_field( $_GET[ $query_var ] ?? '' );
+
+	return \DateTime::createFromFormat( 'Y-m-d', $date ?? '', new \DateTimeZone( 'UTC' ) );
+}
+
+/**
+ * Get the number of unique learners between two dates.
+ *
+ * @param \DateTime $from_date
+ * @param \DateTime $to_date
+ *
+ * @return int
+ */
+function wporg_learn_get_student_count( $from_date, $to_date ) {
+
+	if ( ! $from_date && ! $to_date ) {
```

Comment:
I'm getting an error when filling in only one of the dates.

```
Call to a member function format() on bool
```

Do you intend to make it that both dates are required? If that's the case, maybe the logic should be:

```suggestion
	if ( ! $from_date || ! $to_date ) {
```

WDYT?

Reply:
@m1r0 thanks for the feedback. My focus was on the refactoring to get the php linter to pass, so I wasn't 100% sure myself, happy to make the change.